src/core/authentication/ory.strategy.ts

```diff
@@ -35,31 +36,33 @@ export class OryStrategy extends PassportStrategy(Strategy, 'oathkeeper-jwt') {
     });
   }
 
-  async validate(payload: any) {
+  async validate(payload: KratosPayload) {
     this.logger.verbose?.('Ory Strategy: Kratos payload', LogContext.AUTH);
     this.logger.verbose?.(payload, LogContext.AUTH);
 
-    if (this.checkIfTokenHasExpired(payload.exp))
+    if (checkIfTokenHasExpired(payload.exp))
       throw new TokenException(
         'Access token has expired!',
         AlkemioErrorStatus.TOKEN_EXPIRED
       );
 
-    // Todo: not sure this is correct, but am hitting a case whereby session is null
     let oryIdentity: OryDefaultIdentitySchema | undefined = undefined;
-    if (payload.session) {
-      oryIdentity = payload.session.identity as OryDefaultIdentitySchema;
+    const session = payload.session;
+
+    if (session) {
+      oryIdentity = session.identity as OryDefaultIdentitySchema;
+
+      if (this.authService.shouldExtendSession(session)) {
+        this.authService.extendSession(session);
```

Comment:
```suggestion
        this.authService.extendSession(session).catch(error => {
          this.logger.error(
            `Ory Kratos session failed to be extended: ${error}`,
            LogContext.AUTH
          );
        });
```

That way you will not cancel the flow and will log the error with the correct log format.